src/background_thread.c

```diff
@@ -495,9 +509,22 @@ background_thread_interval_check(tsdn_t *tsdn, arena_t *arena,
 		info->npages_to_purge_new += npurge_new;
 	}
 
-	if (info->npages_to_purge_new > BACKGROUND_THREAD_NPAGES_THRESHOLD ||
-	    (nstime_ns(&info->next_wakeup) ==
-	    BACKGROUND_THREAD_INDEFINITE_SLEEP && info->npages_to_purge_new > 0)) {
+	bool should_signal;
+	if (info->npages_to_purge_new > BACKGROUND_THREAD_NPAGES_THRESHOLD) {
+		should_signal = true;
+	} else if (unlikely(background_thread_indefinite_sleep(info))) {
+		if (extents_npages_get(&arena->extents_dirty) > 0 ||
```

Comment:
Maybe combine this conditional with the enclosing one?  I don't have strong feelings either way.